src/credentials/FabricTable.cpp

```diff
@@ -802,8 +802,13 @@ CHIP_ERROR FabricTable::Delete(FabricIndex fabricIndex)
         // chose to return.
         return CHIP_ERROR_NOT_FOUND;
     }
+
+    // TODO: The error chain below can cause partial state storage. We must refactor.
     ReturnErrorOnFailure(err);
-    ReturnErrorOnFailure(opKeyErr);
+    if (opKeyErr != CHIP_ERROR_INVALID_FABRIC_INDEX)
```

Comment:
Please document that for fail-safe expiration this is an expected condition, because we may already have done `RevertPendingFabricData` in that case.

Reply:
Done